network/nxos/nxos_facts.py

```diff
@@ -16,253 +16,521 @@
 # along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
 #
 
-DOCUMENTATION = '''
+DOCUMENTATION = """
 ---
 module: nxos_facts
 version_added: "2.1"
 short_description: Gets facts about NX-OS switches
 description:
-    - Offers ability to extract facts from device
+  - Collects facts from Cisco Nexus devices running the NX-OS operating
+    system.  Fact collection is supported over both Cli and Nxapi
+    transports.  This module prepends all of the base network fact keys
+    with C(ansible_net_<fact>).  The facts module will always collect a
+    base set of facts from the device and can enable or disable
+    collection of additional facts.
 extends_documentation_fragment: nxos
 author:
     - Jason Edelman (@jedelman8)
     - Gabriele Gerbino (@GGabriele)
-'''
-
-EXAMPLES = '''
-# retrieve facts
-- nxos_facts: host={{ inventory_hostname }}
-'''
-
-RETURN = '''
-facts:
+options:
+  gather_subset:
     description:
-        - Show multiple information about device.
-          These include interfaces, vlans, module and environment information.
-    returned: always
-    type: dict
-    sample: {"fan_info": [{"direction":"front-to-back","hw_ver": "--",
-            "model":"N9K-C9300-FAN2","name":"Fan1(sys_fan1)","status":"Ok"}],
-            "hostname": "N9K2","interfaces": ["mgmt0","Ethernet1/1"],
-            "kickstart": "6.1(2)I3(1)","module": [{"model": "N9K-C9396PX",
-            "ports": "48","status": "active *"}],"os": "6.1(2)I3(1)",
-            "platform": "Nexus9000 C9396PX Chassis","power_supply_info": [{
-            "actual_output": "0 W","model": "N9K-PAC-650W","number": "1",
-            "status":"Shutdown"}],"rr":"Reset Requested by CLI command reload",
-            "vlan_list":[{"admin_state":"noshutdown","interfaces":["Ethernet1/1"],
-            "name": "default","state": "active","vlan_id": "1"}]}
-'''
-
-
-def get_cli_body_ssh(command, response, module):
-    if 'xml' in response[0]:
-        body = []
-    else:
-        body = [json.loads(response[0])]
-    return body
-
-
-def execute_show(cmds, module, command_type=None):
+      - When supplied, this argument will restrict the facts collected
+        to a given subset.  Possible values for this argument include
+        all, hardware, config, legacy, and interfaces.  Can specify a
+        list of values to include a larger subset.  Values can also be used
+        with an initial M(!) to specify that a specific subset should
```

Comment:
`M(!)` will be formatted as a module link, to '!' 

Should this be `C(M(!))` of you want that string to appear?